tendrl/monitoring_integration/grafana/create_dashboards.py

```diff
@@ -53,117 +60,30 @@ def get_cluster_details(integration_id=None):
-                                            "node_details" + str(ex)})
-                    cluster_obj.hosts.append(copy.deepcopy(nodes))
-                except (KeyError, etcd.EtcdKeyNotFound) as ex:
-                        logger.log("error", NS.get("publisher_id", None),
-                                   {'message': "Error while fetching "
-                                    "node id {}".format(node_id) + str(ex)})
-            volume_list = get_resource_keys(cluster_key, "Volumes")
-            for volume_id in volume_list:
-                try:
-                    volume_data = {}
-                    volume_details = etcd_utils.read(cluster_key +
-                                                     "/Volumes/" +
-                                                     str(volume_id))
-                    subvolume_key = cluster_key + "/Volumes/" + str(volume_id)
-                    subvolume_details = get_subvolume_details(subvolume_key)
-                    for volume in volume_details.leaves:
-                        try:
-                            if volume.key.rsplit('/')[-1] in attrs["volumes"]:
-                                volume_data[volume.key.rsplit('/')[-1]] = \
-                                    volume.value
-                        except (KeyError, etcd.EtcdKeyNotFound) as ex:
-                                logger.log("error",
-                                           NS.get("publisher_id", None),
-                                           {'message': "Error while fetching"
-                                            " volume_details" + str(ex)})
-                    volume_data["subvolume"] = subvolume_details
-                    cluster_obj.volumes.append(volume_data)
-                except (KeyError, etcd.EtcdKeyNotFound) as ex:
-                        logger.log("error", NS.get("publisher_id", None),
-                                   {'message': "Error while fetching "
-                                    "volume id {}".format(volume_id) + str(ex)
-                                    })
-            for bricks in cluster_obj.bricks:
-                for volume in cluster_obj.volumes:
-                    try:
-                        if volume["vol_id"] == bricks["vol_id"]:
-                            bricks["vol_name"] = volume["name"]
-                    except (KeyError, etcd.EtcdKeyNotFound) as ex:
-                        logger.log("error", NS.get("publisher_id", None),
-                                   {'message': "Error while getting "
-                                    "volumelevel brick details" + str(ex)})
+            # Get node details
+            cluster_obj.hosts = get_node_details(cluster_key)
+            # Get volume details
+            cluster_obj.volumes = get_volumes_details(cluster_key)
+            # Get brick details from subvolumes
+            cluster_obj.bricks = get_brick_details(cluster_obj.volumes)
```

Comment:
Nice work on simplifying this long method

Reply:
:+1: